services/ymax-planner/src/pending-tx-manager.ts

```diff
@@ -149,15 +180,27 @@ const nobleWithdrawMonitor: PendingTxMonitor<NobleWithdrawTx, EvmContext> = {
       `${logPrefix} Watching Noble withdrawal to ${nobleAddress} for ${amount} ${expectedDenom}`,
     );
 
-    const transferStatus = await watchNobleTransfer({
-      cosmosRest: ctx.cosmosRest,
-      watchAddress: nobleAddress,
-      expectedAmount: amount,
-      expectedDenom,
-      chainKey: 'noble',
-      log: (msg, ...args) => log(`${logPrefix} ${msg}`, ...args),
-      timeoutMs,
-    });
+    let transferStatus = false;
+    if (opts.mode === 'live') {
+      transferStatus = await watchNobleTransfer({
+        cosmosRest: ctx.cosmosRest,
+        watchAddress: nobleAddress,
+        expectedAmount: amount,
+        expectedDenom,
+        chainKey: 'noble',
+        log: (msg, ...args) => log(`${logPrefix} ${msg}`, ...args),
+        timeoutMs: opts.timeoutMs,
+      });
+    } else {
+      transferStatus = await lookBackNobleTransfer({
+        cosmosRest: ctx.cosmosRest,
+        watchAddress: nobleAddress,
+        expectedAmount: amount,
+        expectedDenom,
+        chainKey: 'noble',
+        log: (msg, ...args) => log(`${logPrefix} ${msg}`, ...args),
+      });
+    }
```

Comment:
```suggestion
    const watchArgs = {
      cosmosRest: ctx.cosmosRest,
      watchAddress: nobleAddress,
      expectedAmount: amount,
      expectedDenom,
      chainKey: 'noble',
      log: (msg, ...args) => log(`${logPrefix} ${msg}`, ...args),
    };
    const transferStatus = await (opts.mode === 'live'
      ? watchNobleTransfer({ ...watchArgs, timeoutMs: opts.timeoutMs })
      : lookBackNobleTransfer({ ...watchArgs }));
```
Also, there's no `publishTimeMs` input for `lookBackNobleTransfer`?

Reply:
Currently `lookBackNobleTransfer` works similarly to `watchNobleTransfer`.
It’s a temporary implementation - the long-term goal is to listen to blocks on the Noble chain and extract relevant data with more granular checks (for example, verifying whether CCTP events were emitted).
Because of this, I haven’t included `publishTimeMs` yet, but I’ll add it later if it becomes necessary.